instrumentation-api-incubator/src/main/java/io/opentelemetry/instrumentation/api/incubator/config/internal/ExtendedDeclarativeConfigProperties.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.incubator.config.internal;
+
+import static io.opentelemetry.api.incubator.config.DeclarativeConfigProperties.empty;
+
+import io.opentelemetry.api.incubator.config.DeclarativeConfigProperties;
+import io.opentelemetry.common.ComponentLoader;
+import java.util.List;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * This class is internal and is hence not for public use. Its APIs are unstable and can change at
+ * any time.
+ */
+public final class ExtendedDeclarativeConfigProperties implements DeclarativeConfigProperties {
+
+  private final DeclarativeConfigProperties delegate;
+
+  ExtendedDeclarativeConfigProperties(DeclarativeConfigProperties delegate) {
+    this.delegate = delegate;
+  }
+
+  public ExtendedDeclarativeConfigProperties get(String name) {
```

Comment:
this is the new method in DeclarativeConfigProperties that is proposed in this PR